pkg/discovery/service.go

```
@@ -129,6 +127,10 @@ func (discovery *Discovery) refresh(ctx context.Context) error {
 	}
 
 	for _, node := range list {
+		if ctx.Err() != nil {
```

Just an idea, maybe `Ping` and `UpdateUptime` should check if context was canceled internally?

---

I mean, it should, but I have no idea why doesn't.

---

Ah, found it finally.

---

However, it's still nice to fail early, since we might have a bunch of things in the queue.

---

ah yes you are right, loop is continuing is case of error.

---

https://github.com/storj/storj/pull/1249 should fix the ping not being canceled issue